src/swarm/neo/client/Connection.d

```diff
@@ -462,8 +462,20 @@ public final class Connection: ConnectionBase
         if (!this.restart_after_shutdown)
             return false;
 
-        this.status_ = this.status_.Connecting;
-        retry(this.tryConnect(), this.send_loop, this.epoll);
+        try
+        {
+            this.status_ = this.status_.Connecting;
+            retry(this.tryConnect(), this.send_loop, this.epoll);
+        }
+        catch (TimerException e)
```

Comment:
What was the reason for changing this to only handle that one specific type of exception? My feeling is that we should handle all exceptions here, as we don't want the connection to get into an invalid state.

Reply:
Ah yes, I had thought only about moving the `catch` from `retry` here. But catching everything is better.